hail/src/main/scala/is/hail/expr/types/physical/PCanonicalNDArray.scala

```diff
@@ -91,19 +95,18 @@ final case class PCanonicalNDArray(elementType: PType, nDims: Int, required: Boo
 
   private def getElementAddress(indices: IndexedSeq[Code[Long]], nd: Code[Long], mb: MethodBuilder): Code[Long] = {
     val stridesTuple  = new CodePTuple(strides.pType, strides.load(nd))
-    val bytesAway = mb.newLocal[Long]
+    val runningIndex = mb.newLocal[Long]
     val dataStore = mb.newLocal[Long]
 
-    coerce[Long](Code(
+    Code(
       dataStore := data.load(nd),
-      bytesAway := 0L,
+      runningIndex := 0L,
       indices.zipWithIndex.foldLeft(Code._empty){case (codeSoFar: Code[_], (requestedIndex: Code[Long], strideIndex: Int)) =>
         Code(
           codeSoFar,
-          bytesAway := bytesAway + requestedIndex * stridesTuple(strideIndex))
+          runningIndex := runningIndex + requestedIndex * stridesTuple(strideIndex))
```

Comment:
`strides` is in bytes, so `bytesAway` was a sensible name for this thing. I think that using an index instead is nice since you use that `elementOffset` method you're using below, but you'd have to divide the strides by the elementSize each time, which seems like a bother.